content/en/llm_observability/setup/sdk/python.md

```diff
@@ -0,0 +1,683 @@
+
+`tags`
+: optional - _dictionary_
+<br />A dictionary of string key-value pairs that users can add as tags regarding the evaluation. For more information about tags, see [Getting Started with Tags][9].
+
+### Example
+
+{{< code-block lang="python" >}}
+from ddtrace.llmobs import LLMObs
+from ddtrace.llmobs.decorators import llm
+
+@llm(model_name="claude", name="invoke_llm", model_provider="anthropic")
+def llm_call():
+    completion = ... # user application logic to invoke LLM
+    span_context = LLMObs.export_span(span=None)
+    LLMObs.submit_evaluation(
+        span_context,
+        label="harmfulness",
+        metric_type="score",
+        value=10,
+        tags={"evaluation_provider": "ragas"},
+    )
+    return completion
+{{< /code-block >}}
+
+## Advanced tracing
+
+### Tracing spans using inline methods
+
+For each span kind, the `ddtrace.llmobs.LLMObs` class provides a corresponding inline method to automatically trace the operation a given code block entails. These methods have the same argument signature as their function decorator counterparts, with the addition that `name` defaults to the span kind (`llm`, `workflow`, and so on) if not provided. These methods can be used as context managers to automatically finish the span once the enclosed code block is completed.
+
+#### Example
+
+{{< code-block lang="python" >}}
+from ddtrace.llmobs import LLMObs
+
+def process_message():
+    with LLMObs.workflow(name="process_message", session_id="<SESSION_ID>", ml_app="<ML_APP>") as workflow_span:
+        ... # user application logic
+    return
+{{< /code-block >}}
+
+### Persisting a span across contexts
+
+To manually start and stop a span across different contexts or scopes:
+
+1. Start a span manually using the same methods (for example, the `LLMObs.workflow` method for a workflow span), but as a plain function call rather than as a context manager.
+2. Pass the span object as an argument to other functions.
+3. Stop the span manually with the `span.finish()` method. **Note**: the span must be manually finished, otherwise it will not be submitted.
+
+#### Example
+
+{{< code-block lang="python" >}}
+from ddtrace.llmobs import LLMObs
+
+def process_message():
+    workflow_span = LLMObs.workflow(name="process_message")
+    ... # user application logic
+    separate_task(workflow_span)
+    return
+
+def separate_task(workflow_span):
+    ... # user application logic
+    workflow_span.finish()
+    return
+{{< /code-block >}}
+
+#### Force flushing in serverless environments
+
+`LLMObs.flush()` is a blocking function that submits all buffered LLM Observability data to the Datadog backend. This can be useful in serverless environments to prevent an application from exiting until all LLM Observability traces are submitted.
+
+### Tracing multiple applications
+
+The SDK supports tracking multiple LLM applications from the same service.
```

Comment:
Heading says "tracing," first sentence says "tracking"—is one of these an error?

Reply:
should probably be "tracing" - will update!